relay.go

```diff
@@ -29,6 +30,9 @@ var (
 	RelayAcceptTimeout   = 10 * time.Second
 	HopConnectTimeout    = 30 * time.Second
 	StopHandshakeTimeout = 1 * time.Minute
+
+	HopStreamBuffer = 4096
+	HopStreamLimit  = 1 << 18 // 256K hops for 512K goroutines
```

Comment:
Have we checked this against our current numbers? This seems kind of low, actually. For 20k peers, this'll give us less than 10 streams per peer.

Reply:
We can easily double it -- in fact the mplex relay where I am testing this is running with double the count (it overrides in the daemon init).

Reply:
I'll try with quadruple the count (hop limit at 1M) and evaluate memory usage with it.

Reply:
Seems like we are tight on memory with 2M goroutines active.

Reply:
I doubled the default, and actual daemons can set it higher if they have the resources.